# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/DisplayImpl.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/DisplayImpl.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/DisplayImpl.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/DisplayImpl.h	Fri Nov 08 17:06:00 2019 +0000
@@ -16,10 +16,16 @@
 #include "libANGLE/Stream.h"
 #include "libANGLE/Version.h"
 #include "libANGLE/renderer/EGLImplFactory.h"
+#include "platform/Feature.h"
 
 #include <set>
 #include <vector>
 
+namespace angle
+{
+struct FrontendFeatures;
+}  // namespace angle
+
 namespace egl
 {
 class AttributeMap;
@@ -35,7 +41,7 @@
 namespace gl
 {
 class Context;
-}
+}  // namespace gl
 
 namespace rx
 {
@@ -80,6 +86,7 @@
     virtual egl::Error waitClient(const gl::Context *context)                = 0;
     virtual egl::Error waitNative(const gl::Context *context, EGLint engine) = 0;
     virtual gl::Version getMaxSupportedESVersion() const                     = 0;
+    virtual gl::Version getMaxConformantESVersion() const                    = 0;
     const egl::Caps &getCaps() const;
 
     virtual void setBlobCacheFuncs(EGLSetBlobFuncANDROID set, EGLGetBlobFuncANDROID get) {}
@@ -89,6 +96,12 @@
     void setBlobCache(egl::BlobCache *blobCache) { mBlobCache = blobCache; }
     egl::BlobCache *getBlobCache() const { return mBlobCache; }
 
+    virtual void initializeFrontendFeatures(angle::FrontendFeatures *features) const {}
+
+    virtual void populateFeatureList(angle::FeatureList *features) = 0;
+
+    const egl::DisplayState &getState() const { return mState; }
+
   protected:
     const egl::DisplayState &mState;
 